src/modules/ShopgateAppCodeExecutor.js

```diff
@@ -0,0 +1,67 @@
+import { executeWithRetry } from './executeWithRetry';
+import { isShopgateJSBridgeAvailable } from './isShopgateJSBridgeAvailable';
+
+/**
+ * This class takes care if Shopgate App related code should be executed or not
+ * depending on the environment
+ */
+export class ShopgateAppCodeExecutor {
+  /**
+   * Constructor
+   */
+  constructor() {
+    this.shopgateApp = null;
+    this.evaluate = false;
+    this.callbacks = [];
+  }
+
+  /**
+   * Tries to execute a Shopgate App related code
+   *
+   * @param {function} callback the method to call if we are executed inside of a Shopgate App
+   */
+  execute(callback) {
+    if (!this.shopgateApp) {
+      this.callbacks.push(callback);
+
+      if (this.shopgateApp === null) {
+        this.evaluateShopgateApp();
```

Comment:
If we move the evaluation to the constructor, we'll no longer need an if statement here and could simplify evaluation process too.